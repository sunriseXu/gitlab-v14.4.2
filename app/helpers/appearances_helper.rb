# frozen_string_literal: true

module AppearancesHelper
  include MarkupHelper
  include Gitlab::Utils::StrongMemoize

  def brand_title
    current_appearance&.title.presence || default_brand_title
  end

  def default_brand_title
    # This resides in a separate method so that EE can easily redefine it.
    'GitLab Community Edition'
  end

  def brand_image
    image_tag(current_appearance.logo_path) if current_appearance&.logo?
  end

  def brand_text
    markdown_field(current_appearance, :description)
  end

  def brand_new_project_guidelines
    markdown_field(current_appearance, :new_project_guidelines)
  end

  def brand_profile_image_guidelines
    markdown_field(current_appearance, :profile_image_guidelines)
  end

  def current_appearance
    strong_memoize(:current_appearance) do
      Appearance.current
    end
  end

  def brand_header_logo(options = {})
    add_gitlab_white_text = options[:add_gitlab_white_text] || false
    add_gitlab_black_text = options[:add_gitlab_black_text] || false

    if current_appearance&.header_logo?
      image_tag current_appearance.header_logo_path, class: 'brand-header-logo'
    elsif add_gitlab_white_text
      render partial: 'shared/logo_with_white_text', formats: :svg
    elsif add_gitlab_black_text
      render partial: 'shared/logo_with_black_text', formats: :svg
    else
      render partial: 'shared/logo', formats: :svg
    end
  end

  def header_message
    return unless current_appearance&.show_header?

    class_names = []
    class_names << 'with-performance-bar' if performance_bar_enabled?

    render_message(:header_message, class_names: class_names)
  end

  def footer_message
    return unless current_appearance&.show_footer?

    render_message(:footer_message)
  end

  private

  def render_message(field_sym, class_names: [], style: message_style)
    class_names << field_sym.to_s.dasherize

    content_tag :div, class: class_names, style: style do
      markdown_field(current_appearance, field_sym)
    end
  end

  def message_style
    style = []
    style << "background-color: #{current_appearance.message_background_color};"
    style << "color: #{current_appearance.message_font_color}"
    style.join
  end
end

AppearancesHelper.prepend_mod
