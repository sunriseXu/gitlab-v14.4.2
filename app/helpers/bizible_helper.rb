# frozen_string_literal: true

module BizibleHelper
  def bizible_enabled?
    Feature.enabled?(:ecomm_instrumentation, type: :ops) &&
      Gitlab.config.extra.has_key?('bizible') &&
      Gitlab.config.extra.bizible.present? &&
      Gitlab.config.extra.bizible == true
  end
end
