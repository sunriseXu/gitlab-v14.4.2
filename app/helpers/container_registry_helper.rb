# frozen_string_literal: true

module ContainerRegistryHelper
  def container_repository_gid_prefix
    "gid://#{GlobalID.app}/#{ContainerRepository.name}/"
  end
end
