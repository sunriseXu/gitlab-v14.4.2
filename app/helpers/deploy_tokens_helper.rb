# frozen_string_literal: true

module DeployTokensHelper
  def expand_deploy_tokens_section?(new_deploy_token, created_deploy_token)
    created_deploy_token ||
      new_deploy_token.errors.present? ||
      Rails.env.test?
  end

  def container_registry_enabled?(group_or_project)
    Gitlab.config.registry.enabled &&
      can?(current_user, :read_container_image, group_or_project)
  end

  def packages_registry_enabled?(group_or_project)
    Gitlab.config.packages.enabled &&
      can?(current_user, :read_package, group_or_project&.packages_policy_subject)
  end

  def deploy_token_revoke_button_data(token:, group_or_project:)
    {
      token: token.to_json(only: [:id, :name]),
      revoke_path: revoke_deploy_token_path(group_or_project, token)
    }
  end
end
