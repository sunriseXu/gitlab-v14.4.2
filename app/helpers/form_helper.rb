# frozen_string_literal: true

module FormHelper
  def form_errors(model, type: 'form', truncate: [])
    errors = model.errors

    return unless errors.any?

    headline = n_(
      'The %{type} contains the following error:',
      'The %{type} contains the following errors:',
      errors.count
    ) % { type: type }

    truncate = Array.wrap(truncate)

    messages = errors.map do |error|
      attribute = error.attribute
      message = error.message

      message = html_escape_once(errors.full_message(attribute, message)).html_safe
      message = tag.span(message, class: 'str-truncated-100') if truncate.include?(attribute)
      message = append_help_page_link(message, error.options) if error.options[:help_page_url].present?

      tag.li(message)
    end.join.html_safe

    render Pajamas::AlertComponent.new(
      variant: :danger,
      title: headline,
      dismissible: false,
      alert_options: { id: 'error_explanation', class: 'gl-mb-5' }
    ) do |c|
      c.body do
        tag.ul(class: 'gl-pl-5 gl-mb-0') do
          messages
        end
      end
    end
  end

  def dropdown_max_select(data)
    return data[:'max-select'] unless Feature.enabled?(:limit_reviewer_and_assignee_size)

    if data[:'max-select'] && data[:'max-select'] < MergeRequest::MAX_NUMBER_OF_ASSIGNEES_OR_REVIEWERS
      data[:'max-select']
    else
      MergeRequest::MAX_NUMBER_OF_ASSIGNEES_OR_REVIEWERS
    end
  end

  def assignees_dropdown_options(issuable_type)
    dropdown_data = {
      toggle_class: 'js-user-search js-assignee-search js-multiselect js-save-user-data',
      title: _('Select assignee'),
      filter: true,
      dropdown_class: 'dropdown-menu-user dropdown-menu-selectable dropdown-menu-assignee',
      placeholder: _('Search users'),
      data: {
        first_user: current_user&.username,
        null_user: true,
        current_user: true,
        project_id: (@target_project || @project)&.id,
        field_name: "#{issuable_type}[assignee_ids][]",
        default_label: _('Unassigned'),
        'max-select': 1,
        'dropdown-header': s_('SearchToken|Assignee'),
        multi_select: true,
        'input-meta': 'name',
        'always-show-selectbox': true,
        current_user_info: UserSerializer.new.represent(current_user)
      }
    }

    type = issuable_type.to_s

    if type == 'issue' && issue_supports_multiple_assignees? ||
        type == 'merge_request' && merge_request_supports_multiple_assignees?
      dropdown_data = multiple_assignees_dropdown_options(dropdown_data)
    end

    dropdown_data
  end

  def reviewers_dropdown_options(issuable_type, iid = nil, target_branch = nil)
    dropdown_data = {
      toggle_class: 'js-reviewer-search js-multiselect js-save-user-data',
      title: 'Request review from',
      filter: true,
      dropdown_class: 'dropdown-menu-user dropdown-menu-selectable dropdown-menu-reviewer',
      placeholder: _('Search users'),
      data: {
        first_user: current_user&.username,
        null_user: true,
        current_user: true,
        project_id: (@target_project || @project)&.id,
        field_name: "#{issuable_type}[reviewer_ids][]",
        default_label: 'Unassigned',
        'max-select': 1,
        'dropdown-header': 'Reviewer',
        multi_select: true,
        'input-meta': 'name',
        'always-show-selectbox': true,
        current_user_info: UserSerializer.new.represent(current_user)
      }
    }

    if iid
      dropdown_data[:data][:iid] = iid
    end

    if target_branch
      dropdown_data[:data][:target_branch] = target_branch
    end

    if merge_request_supports_multiple_reviewers?
      dropdown_data = multiple_reviewers_dropdown_options(dropdown_data)
    end

    dropdown_data
  end

  # Overwritten
  def issue_supports_multiple_assignees?
    false
  end

  # Overwritten
  def merge_request_supports_multiple_assignees?
    false
  end

  # Overwritten
  def merge_request_supports_multiple_reviewers?
    false
  end

  private

  def append_help_page_link(message, options)
    help_page_url = options[:help_page_url]
    help_link_text = options[:help_link_text] || _('Learn more.')

    help_link = link_to(
      help_link_text,
      help_page_url,
      target: '_blank',
      rel: 'noopener noreferrer'
    )

    message + " #{help_link}".html_safe
  end

  def multiple_assignees_dropdown_options(options)
    new_options = options.dup

    new_options[:title] = _('Select assignee(s)')
    new_options[:data][:'dropdown-header'] = 'Assignee(s)'
    new_options[:data].delete(:'max-select')

    new_options
  end

  def multiple_reviewers_dropdown_options(options)
    new_options = options.dup

    new_options[:title] = _('Select reviewer(s)')
    new_options[:data][:'dropdown-header'] = _('Reviewer(s)')

    if Feature.enabled?(:limit_reviewer_and_assignee_size)
      new_options[:data][:'max-select'] = MergeRequest::MAX_NUMBER_OF_ASSIGNEES_OR_REVIEWERS
    else
      new_options[:data].delete(:'max-select')
    end

    new_options
  end
end

FormHelper.prepend_mod_with('FormHelper')
