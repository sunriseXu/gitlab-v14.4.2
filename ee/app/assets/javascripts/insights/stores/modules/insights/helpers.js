import { CHART_TYPES } from 'ee/insights/constants';
import { __, s__ } from '~/locale';

const getGroupByValue = (queryParams) => {
  return queryParams.group_by || queryParams.params.group_by;
};

const getTypeValue = (queryParams) => {
  return queryParams.issuable_type || queryParams.params.issuable_type || queryParams.params.metric;
};

const getAxisTitle = (label) => {
  switch (label) {
    case 'day':
      return __('Days');
    case 'week':
      return __('Weeks');
    case 'month':
      return __('Months');
    case 'issue':
      return __('Issues');
    case 'merge_request':
      return __('Merge requests');
    case 'deployment_frequency':
      return s__('DORA4Metrics|Deployment frequency');
    case 'lead_time_for_changes':
      return s__('DORA4Metrics|Lead time for changes (median days)');
    case 'time_to_restore_service':
      return s__('DORA4Metrics|Time to restore service (median days)');
    case 'change_failure_rate':
      return s__('DORA4Metrics|Change failure rate (percentage)');
    default:
      return '';
  }
};

export const transformChartDataForGlCharts = (
  { type, query: queryParams },
  { labels, datasets },
) => {
  const formattedData = {
    xAxisTitle: getAxisTitle(getGroupByValue(queryParams)),
    yAxisTitle: getAxisTitle(getTypeValue(queryParams)),
    labels,
    datasets: [],
    seriesNames: [],
  };

  switch (type) {
    case CHART_TYPES.BAR:
      formattedData.datasets = [
        {
          name: 'all',
          data: labels.map((label, i) => [label, datasets[0].data[i]]),
        },
      ];
      break;
    case CHART_TYPES.STACKED_BAR:
      formattedData.datasets.push(
        ...datasets.map((dataset) => ({
          name: dataset.label,
          data: dataset.data,
        })),
      );
      break;
    case CHART_TYPES.LINE:
      formattedData.datasets.push(
        ...datasets.map((dataset) => ({
          name: dataset.label,
          data: labels.map((label, i) => [label, dataset.data[i]]),
        })),
      );

      break;
    default:
      formattedData.datasets = { all: labels.map((label, i) => [label, datasets[0].data[i]]) };
  }
  return formattedData;
};

export default {
  transformChartDataForGlCharts,
};
