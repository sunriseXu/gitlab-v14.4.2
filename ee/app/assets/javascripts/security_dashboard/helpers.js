import isPlainObject from 'lodash/isPlainObject';
import groupBy from 'lodash/groupBy';
import {
  REPORT_TYPES_WITH_MANUALLY_ADDED,
  REPORT_TYPES_WITH_CLUSTER_IMAGE,
  REPORT_TYPES_ALL,
  SEVERITY_LEVELS,
} from 'ee/security_dashboard/store/constants';
import convertReportType from 'ee/vue_shared/security_reports/store/utils/convert_report_type';
import { VULNERABILITY_STATES } from 'ee/vulnerabilities/constants';
import { convertObjectPropsToSnakeCase } from '~/lib/utils/common_utils';
import { s__, __ } from '~/locale';
import { SCANNER_NAMES_MAP } from '~/security_configuration/components/constants';
import { DEFAULT_SCANNER } from './constants';

const parseOptions = (obj) =>
  Object.entries(obj).map(([id, name]) => ({ id: id.toUpperCase(), name }));

const toolName = s__('SecurityReports|Tool');
const allToolsName = s__('ciReport|All tools');
const stateOptions = parseOptions(VULNERABILITY_STATES);
const defaultStateOptions = stateOptions.filter((x) => ['DETECTED', 'CONFIRMED'].includes(x.id));

export const stateFilter = {
  name: s__('SecurityReports|Status'),
  id: 'state',
  options: stateOptions,
  allOption: { id: 'all', name: s__('VulnerabilityStatusTypes|All statuses') },
  defaultOptions: defaultStateOptions,
};

export const severityFilter = {
  name: s__('SecurityReports|Severity'),
  id: 'severity',
  options: parseOptions(SEVERITY_LEVELS),
  allOption: { name: s__('ciReport|All severities') },
  defaultOptions: [],
};

export const createScannerOption = (vendor, reportType) => {
  const type = reportType.toUpperCase();

  return {
    id: `${vendor}.${type}`,
    reportType: reportType.toUpperCase(),
    name: convertReportType(reportType),
    scannerIds: [],
  };
};

// This is used on the pipeline security tab, group-level report, and instance-level report. It's
// used by the scanner filter that shows a flat list of scan types (DAST, SAST, etc) with no vendor
// grouping.
export const simpleScannerFilter = {
  name: toolName,
  id: 'reportType',
  options: parseOptions(REPORT_TYPES_WITH_MANUALLY_ADDED),
  allOption: { name: allToolsName },
  defaultOptions: [],
};

export const simpleScannerFilterPipeline = {
  name: toolName,
  id: 'reportType',
  options: parseOptions(REPORT_TYPES_WITH_CLUSTER_IMAGE),
  allOption: { name: allToolsName },
  defaultOptions: [],
};

// This is used on the project-level report. It's used by the scanner filter that shows a list of
// scan types (DAST, SAST, etc) that's grouped by vendor.
export const vendorScannerFilter = {
  name: toolName,
  id: 'scanner',
  options: Object.keys(REPORT_TYPES_WITH_MANUALLY_ADDED).map((x) =>
    createScannerOption(DEFAULT_SCANNER, x),
  ),
  allOption: { name: allToolsName },
  defaultOptions: [],
};

export const activityOptions = {
  NO_ACTIVITY: { id: 'NO_ACTIVITY', name: s__('SecurityReports|No activity') },
  WITH_ISSUES: { id: 'WITH_ISSUES', name: s__('SecurityReports|With issues') },
  NO_LONGER_DETECTED: { id: 'NO_LONGER_DETECTED', name: s__('SecurityReports|No longer detected') },
};

export const activityFilter = {
  name: s__('Reports|Activity'),
  id: 'activity',
  options: Object.values(activityOptions),
  allOption: { name: s__('SecurityReports|All activity') },
  defaultOptions: [],
};

export const projectFilter = {
  name: s__('SecurityReports|Project'),
  id: 'projectId',
  options: [],
  allOption: { name: s__('ciReport|All projects') },
  defaultOptions: [],
};

export const clusterFilter = {
  name: s__('SecurityReports|Cluster'),
  id: 'cluster',
  options: [],
  allOption: { name: s__('ciReport|All clusters') },
  defaultOptions: [],
};

export const imageFilter = {
  name: s__('SecurityReports|Image'),
  id: 'image',
  options: [],
  allOption: { name: s__('ciReport|All images') },
  defaultOptions: [],
};

/**
 * Provided a security reports summary from the GraphQL API, this returns an array of arrays
 * representing a properly formatted report ready to be displayed in the UI. Each sub-array consists
 * of the user-friend report's name, and the summary's payload. Note that summary entries are
 * considered empty and are filtered out of the return if the payload is `null` or don't include
 * a vulnerabilitiesCount property. Report types whose name can't be matched to a user-friendly
 * name are filtered out as well.
 *
 * Take the following summary for example:
 * {
 *   containerScanning: { vulnerabilitiesCount: 123 },
 *   invalidReportType: { vulnerabilitiesCount: 123 },
 *   dast: null,
 * }
 *
 * The formatted summary would look like this:
 * [
 *   ['containerScanning', { vulnerabilitiesCount: 123 }]
 * ]
 *
 * Note that `invalidReportType` was filtered out as it can't be matched with a user-friendly name,
 * and the DAST report was omitted because it's empty (`null`).
 *
 * @param {Object} rawSummary
 * @returns {Array}
 */
export const getFormattedSummary = (rawSummary = {}) => {
  if (!isPlainObject(rawSummary)) {
    return [];
  }
  // Convert keys to snake case so they can be matched against REPORT_TYPES keys for translation
  const snakeCasedSummary = convertObjectPropsToSnakeCase(rawSummary);
  // Convert object to an array of entries to make it easier to loop through
  const summaryEntries = Object.entries(snakeCasedSummary);
  // Filter out empty entries as we don't want to display those in the summary
  const withoutEmptyEntries = summaryEntries.filter(
    ([, scanSummary]) => scanSummary?.vulnerabilitiesCount !== undefined,
  );
  // Replace keys with translations found in REPORT_TYPES if available
  const formattedEntries = withoutEmptyEntries.map(([scanType, scanSummary]) => {
    const name = REPORT_TYPES_ALL[scanType];
    return name ? [name, scanSummary] : null;
  });
  // Filter out keys that could not be matched with any translation and are thus considered invalid
  return formattedEntries.filter((entry) => entry !== null);
};

/**
 * We have disabled loading hasNextPage from GraphQL as it causes timeouts in database,
 * instead we have to calculate that value based on the existence of endCursor. When endCursor
 * is empty or has null value, that means that there is no next page to be loaded from GraphQL API.
 *
 * @param {Object} pageInfo
 * @returns {Object}
 */
export const preparePageInfo = (pageInfo) => {
  return { ...pageInfo, hasNextPage: Boolean(pageInfo?.endCursor) };
};

/**
 * Provided a vulnerability scanners from the GraphQL API, this returns an array that is
 * formatted so it can be displayed in the dropdown UI.
 *
 * @param {Array} vulnerabilityScanners
 * @returns {Array} formatted vulnerabilityScanners
 */
export const getFormattedScanners = (vulnerabilityScanners) => {
  const groupedByReportType = groupBy(vulnerabilityScanners, 'reportType');

  return Object.entries(groupedByReportType).map(([reportType, scanners]) => {
    return {
      id: reportType,
      reportType,
      name: SCANNER_NAMES_MAP[reportType] || SCANNER_NAMES_MAP.GENERIC,
      scannerIds: scanners.map(({ id }) => id),
    };
  });
};

export const PROJECT_LOADING_ERROR_MESSAGE = __('An error occurred while retrieving projects.');

export default () => ({});
