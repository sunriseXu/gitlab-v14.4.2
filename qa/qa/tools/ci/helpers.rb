# frozen_string_literal: true

module QA
  module Tools
    module Ci
      # Helpers for CI related tasks
      #
      module Helpers
        include Support::API

        # Logger instance
        #
        # @return [Logger]
        def logger
          @logger ||= Gitlab::QA::TestLogger.logger(
            level: Gitlab::QA::Runtime::Env.log_level,
            source: "CI Tools"
          )
        end

        # Api get request
        #
        # @param [String] path
        # @param [Hash] args
        # @return [Hash, Array]
        def api_get(path, **args)
          response = get("#{api_url}/#{path}", { headers: { "PRIVATE-TOKEN" => access_token }, **args })
          response = response.follow_redirection if response.code == Support::API::HTTP_STATUS_PERMANENT_REDIRECT
          raise "Request failed: '#{response.body}'" unless response.code == Support::API::HTTP_STATUS_OK

          args[:raw_response] ? response : parse_body(response)
        end

        # Gitlab api url
        #
        # @return [String]
        def api_url
          @api_url ||= ENV.fetch('CI_API_V4_URL', 'https://gitlab.com/api/v4')
        end

        # Api access token
        #
        # @return [String]
        def access_token
          @access_token ||= ENV.fetch('QA_GITLAB_CI_TOKEN') { raise('Variable QA_GITLAB_CI_TOKEN missing') }
        end
      end
    end
  end
end
