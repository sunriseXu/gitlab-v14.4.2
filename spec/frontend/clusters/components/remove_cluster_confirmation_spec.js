import { GlModal, GlSprintf } from '@gitlab/ui';
import { mount } from '@vue/test-utils';
import { nextTick } from 'vue';
import { stubComponent } from 'helpers/stub_component';
import RemoveClusterConfirmation from '~/clusters/components/remove_cluster_confirmation.vue';

describe('Remove cluster confirmation modal', () => {
  let wrapper;

  const createComponent = ({ props = {}, stubs = {} } = {}) => {
    wrapper = mount(RemoveClusterConfirmation, {
      propsData: {
        clusterPath: 'clusterPath',
        clusterName: 'clusterName',
        ...props,
      },
      stubs,
    });
  };

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  it('renders buttons with modal included', () => {
    createComponent();
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('two buttons', () => {
    const findModal = () => wrapper.findComponent(GlModal);
    const findRemoveIntegrationButton = () =>
      wrapper.find('[data-testid="remove-integration-button"]');
    const findRemoveIntegrationAndResourcesButton = () =>
      wrapper.find('[data-testid="remove-integration-and-resources-button"]');

    beforeEach(() => {
      createComponent({
        props: { clusterName: 'my-test-cluster' },
        stubs: { GlSprintf, GlModal: stubComponent(GlModal) },
      });
      jest.spyOn(findModal().vm, 'show').mockReturnValue();
    });

    it('open modal with "cleanup" option', async () => {
      findRemoveIntegrationAndResourcesButton().trigger('click');

      await nextTick();

      expect(findModal().vm.show).toHaveBeenCalled();
      expect(wrapper.vm.confirmCleanup).toEqual(true);
      expect(findModal().html()).toContain(
        '<strong>To remove your integration and resources, type <code>my-test-cluster</code> to confirm:</strong>',
      );
    });

    it('open modal without "cleanup" option', async () => {
      findRemoveIntegrationButton().trigger('click');

      await nextTick();

      expect(findModal().vm.show).toHaveBeenCalled();
      expect(wrapper.vm.confirmCleanup).toEqual(false);
      expect(findModal().html()).toContain(
        '<strong>To remove your integration, type <code>my-test-cluster</code> to confirm:</strong>',
      );
    });

    describe('with cluster management project', () => {
      beforeEach(() => {
        createComponent({ props: { hasManagementProject: true } });
      });

      it('renders regular button instead', () => {
        expect(findRemoveIntegrationAndResourcesButton().exists()).toBe(false);
        expect(findRemoveIntegrationButton().exists()).toBe(true);
      });
    });
  });
});
