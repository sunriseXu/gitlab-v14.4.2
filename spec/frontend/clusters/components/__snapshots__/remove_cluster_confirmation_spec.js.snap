// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Remove cluster confirmation modal renders buttons with modal included 1`] = `
<div
  class="gl-display-flex"
>
  <button
    class="btn gl-mr-3 btn-danger btn-md gl-button"
    data-testid="remove-integration-and-resources-button"
    type="button"
  >
    <!---->
     
    <!---->
      
    <span
      class="gl-button-text"
    >
      
    Remove integration and resources
  
    </span>
  </button>
   
  <button
    class="btn btn-danger btn-md gl-button btn-danger-secondary"
    data-testid="remove-integration-button"
    type="button"
  >
    <!---->
     
    <!---->
      
    <span
      class="gl-button-text"
    >
      
    Remove integration
  
    </span>
  </button>
   
  <!---->
</div>
`;
